whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/NrtmVersionInfoRepository.java

```diff
@@ -70,7 +76,12 @@ public List<NrtmVersionInfo> findLastVersionPerSource() {
         }
     }
 
-    public NrtmVersionInfo findLastSnapshotVersionForSource(final NrtmSourceModel source) {
+    public Optional<NrtmVersionInfo> findLastVersion(final NrtmSource source) {
+       return findLastVersionPerSource().stream().filter( (versionInfo) ->  versionInfo.source().getName().equals(source.getName())).findFirst();
```

Comment:
Here we are calling findLastVersionPerSource and returning the data per each source, when we just want the data for one of them. is it not better to perform the query just adding the condition of the source there?